### PR TITLE
 Fix not-useful capabilities unit tests 

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
Make the Capabilities unit tests assert things about array keys when it is array keys they care about.

## Related Issue
#31588 

## Motivation and Context
Some of the Capabilities unit tests are rubbish - they are checking for values in the array rather than that keys exist. The test assertions do not do what you think they do!

See #31568 for a demonstration of what rubbish can pass.

## How Has This Been Tested?
Run unit tests locally.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Unit test "bugfix"

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
(and pass even better)

### CODE REVIEW

**Comment by phil-davis** [comment]
And while were are here, fix typo ``incomming`` => ``incoming`` everywhere I could find it.

File: apps/files_sharing/tests/CapabilitiesTest.php [hunk 1]
```diff
@@ -83,9 +83,9 @@ public function testEnabledSharingAPI() {
 		];
 		$result = $this->getResults($map);
 		$this->assertTrue($result['api_enabled']);
-		$this->assertContains('public', $result);
-		$this->assertContains('user', $result);
-		$this->assertContains('resharing', $result);
+		$this->assertArrayHasKey('public', $result);
+		$this->assertArrayHasKey('user', $result);
+		$this->assertArrayHasKey('resharing', $result);
```
Comment (phil-davis) [inline apps/files_sharing/tests/CapabilitiesTest.php]: Locally I tried changing each of the expected key strings to confirm the test really can fail.

File: apps/files_sharing/tests/CapabilitiesTest.php [hunk 2]
```diff
@@ -94,9 +94,9 @@ public function testDisabledSharingAPI() {
 		];
 		$result = $this->getResults($map);
 		$this->assertFalse($result['api_enabled']);
-		$this->assertNotContains('public', $result);
-		$this->assertNotContains('user', $result);
-		$this->assertNotContains('resharing', $result);
+		$this->assertFalse($result['public']['enabled']);
+		$this->assertFalse($result['user']['send_mail']);
+		$this->assertFalse($result['resharing']);
```
Comment (phil-davis) [inline apps/files_sharing/tests/CapabilitiesTest.php]: Locally I tried changing each of the expected key strings to confirm the test really can fail.

**Comment by phil-davis** [comment]
We need unit and acceptance tests that test the unit and acceptance tests [ that test the unit and acceptance tests...]

**Review by PVince81 — APPROVED** [review]
:+1: please backport

**Comment by phil-davis** [comment]
Backport ``stable10`` #31587

**Pull request merged**